### PR TITLE
TEIIDSB 211 spring data s3

### PR DESCRIPTION
**Pull request by aditya300899**

### CODE REVIEW

File: data/s3/src/main/java/org/teiid/spring/data/s3/S3Configuration.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.teiid.spring.data.s3;
+
+public class S3Configuration {
+
+    private String awsAccessKey;
```
Comment (rareddy) [inline data/s3/src/main/java/org/teiid/spring/data/s3/S3Configuration.java:22]: Can we change properties to match current amazon-s3? I mean remove "was" from "awsAccessKey"

Comment (aditya300899) [reply data/s3/src/main/java/org/teiid/spring/data/s3/S3Configuration.java:22]: I've checked it with the s3executionfactory, it is "awsAccessKey" and not "accessKey" if you are saying about the field name.

**Comment by aditya300899** [comment]
> In the existing implementation that's only on getFile. It must happen automatically on save on the server side.

Like the getObject method takes a getObjectRequest, likewise the putObject also takes a putObjectRequest where we can set the key and corresponding algorithm!

I've added the functionality at both the places!

File: data/s3/src/main/java/org/teiid/spring/data/s3/S3Configuration.java (hunk 1 above)
Comment (shawkins) [reply data/s3/src/main/java/org/teiid/spring/data/s3/S3Configuration.java:22]: I'd vote for removing the aws prefix regardless.

Comment (rareddy) [reply data/s3/src/main/java/org/teiid/spring/data/s3/S3Configuration.java:22]: +1

**Comment by aditya300899** [comment]
@shawkins I spent the day pretty much figuring out how to test using mockito. In the testGetFiles(), I was getting this error:
org.mockito.exceptions.misusing.MissingMethodInvocationException: 
when() requires an argument which has to be 'a method call on a mock'.
For example:
    when(mock.getArticles()).thenReturn(articles);

Also, this error might show up because:
1. you stub either of: final/private/equals()/hashCode() methods.
   Those methods *cannot* be stubbed/verified.
   Mocking methods declared on non-public parent classes is not supported.
2. inside when() you don't call method on mock but on some other object.

Can you help me, what I'm doing wrong?

And also, see that If I've structured the test well or it is supposed to be some other way?

**Comment by aditya300899** [comment]
I've figured out the problem. I think it's good now. Suggest changes if any!!

**Comment by shawkins** [comment]
@aditya300899 were there any additional commits that you were going to make, such as minor cleanups of the tests like removing Assert.assertTrue(true), etc.?  I can also do that with the porting to WildFly if you haven't already made those changes.

**Comment by aditya300899** [comment]
I thought maybe you'd have more suggestions later, thus did not commit. Also, I was to think about the code coverage of a test by using some support in IntelliJ! I would do this later now. 
As far as I remember and I'd noted down there were two changes:
1. remove the AssertTrue(true)
2. and not mocking the configuration
I've included those changes in the commit above.

**Pull request merged**